### PR TITLE
Teams Interop feature merge

### PR DESCRIPTION
**Pull request by iaulakh**
## Purpose

* Teams Interop feature merge

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

### CODE REVIEW

**Review by lpeixin — DISMISSED** [review]

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java [hunk 1]
```diff
@@ -363,46 +467,52 @@ private void openShareDialogue() {
         Log.d(LOG_TAG, "Share button clicked!");
         final Intent sendIntent = new Intent();
         sendIntent.setAction(Intent.ACTION_SEND);
-        sendIntent.putExtra(Intent.EXTRA_TEXT, callingContext.getGroupId());
+        sendIntent.putExtra(Intent.EXTRA_TEXT, callingContext.getJoinId());
```
Comment (waynemo) [inline AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java:470]: We should consider applying these renaming changes in main branch first to minimize changes later on when merging.

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/EndActivity.java [hunk 2]
```diff
@@ -37,6 +38,28 @@ protected void onCreate(final Bundle savedInstanceState) {
...
+    @Override
+    public void onBackPressed() {
+        super.onBackPressed();
+        startIntroActivity();
+    }
+
+    @Override
+    protected void onDestroy() {
+        Log.d(LOG_TAG, "EndActivity - onDestroy");
+        super.onDestroy();
```
Comment (waynemo) [inline AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/EndActivity.java:62]: These appear to be changes that are not directly related to Teams Interop. Should they be applied to the main branch first?

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/JoinCallActivity.java [hunk 3]
```diff
@@ -41,6 +54,13 @@ protected void onCreate(final Bundle savedInstanceState) {
         }
 
         initializeUI();
+
+        mStartForResult = registerForActivityResult(new ActivityResultContracts.StartActivityForResult(),
+            result -> {
+                if (result.getResultCode() == Activity.RESULT_OK) {
+                    finish();
+                }
+            });
```
Comment (waynemo) [inline AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/JoinCallActivity.java:63]: What is this for?

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/SetupActivity.java [hunk 4]
```diff
@@ -178,7 +182,8 @@ private void openSettings() {
         intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
         final Uri uri = Uri.fromParts("package", getPackageName(), null);
         intent.setData(uri);
-        finishAffinity();
+        setResult(RESULT_OK);
+        finish();
```
Comment (waynemo) [inline AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/SetupActivity.java:186]: What is this for?

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/SetupActivity.java [hunk 5]
```diff
@@ -85,7 +88,8 @@ protected void onCreate(final Bundle savedInstanceState) {
         final CompletableFuture<Void> setupCompletableFuture = callingContext.setupAsync();
 
         final Intent intent = getIntent();
-        groupId = intent.getStringExtra(Constants.GROUP_ID);
+        callType = (JoinCallType) intent.getSerializableExtra(Constants.CALL_TYPE);
+        joinId = intent.getStringExtra(Constants.JOIN_ID);
```
Comment (waynemo) [inline AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/SetupActivity.java:92]: We should consider applying these renaming changes in main branch first to minimize changes later on when merging.

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/external/calling/CallingContext.java [hunk 6]
```diff
@@ -56,7 +58,7 @@
     Context appContext;
 
     private final Callable<String> tokenFetcher;
-    private String groupId;
+    private String joinId;
```
Comment (waynemo) [inline AzureCalling/app/src/main/java/com/azure/samples/communication/calling/external/calling/CallingContext.java:61]: We should consider applying these renaming changes in main branch first to minimize changes later on when merging.

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/external/calling/JoinCallConfig.java [hunk 7]
```diff
@@ -3,27 +3,32 @@
...
         this.isMicrophoneMuted = isMicrophoneMuted;
         this.isCameraOn = isCameraOn;
         this.displayName = displayName;
+        this.callType = callType;
     }
 
-    public String getGroupId() {
-        return groupId;
+    public String getJoinId() {
+        return joinId;
```
Comment (waynemo) [inline AzureCalling/app/src/main/java/com/azure/samples/communication/calling/external/calling/JoinCallConfig.java:31]: We should consider applying these renaming changes in main branch first to minimize changes later on when merging.

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/helpers/Constants.java [hunk 8]
```diff
@@ -12,9 +12,12 @@
...
 
     /* The key for the meeting name */
-    public static final String GROUP_ID = "GROUP_ID";
+    public static final String JOIN_ID = "JOIN_ID";
 
     /* The key for the Join Call Config */
     public static final String JOIN_CALL_CONFIG = "JOIN_CALL_CONFIG";
 
+    /* The key for the Join Call Type */
+    public static final String CALL_TYPE = "CALL_TYPE";
```
Comment (waynemo) [inline AzureCalling/app/src/main/java/com/azure/samples/communication/calling/helpers/Constants.java:21]: We should consider applying these renaming changes in main branch first to minimize changes later on when merging.

File: AzureCalling/app/src/main/res/layout/activity_call.xml [hunk 9]
```diff
@@ -113,48 +118,90 @@
...
+            android:textSize="20sp" />
+
+    </LinearLayout>
+
     <LinearLayout
         android:id="@+id/call_hangup_overlay"
         android:layout_width="0dp"
         android:layout_height="0dp"
-        android:background="#66000000"
+        android:background="#CC000000"
```
Comment (waynemo) [inline AzureCalling/app/src/main/res/layout/activity_call.xml:166]: Is this a change that would be applicable in the main branch as well?

**Review by waynemo — CHANGES_REQUESTED** [review]

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/JoinCallActivity.java (hunk 3 above)
Comment (iaulakh) [reply AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/JoinCallActivity.java:63]: once AndroidSettings are opened, we need to go back to InroActivity.

File: AzureCalling/app/src/main/res/layout/activity_call.xml (hunk 9 above)
Comment (iaulakh) [reply AzureCalling/app/src/main/res/layout/activity_call.xml:166]: yes, sent new PR to main.

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/SetupActivity.java [hunk 10]
```diff
@@ -241,7 +241,6 @@ private void startCall() {
                 final JoinCallConfig joinCallConfig = new JoinCallConfig(
                         joinId, !audioToggleButton.isChecked(), videoToggleButton.isChecked(),
                         setupName.getText().toString(), callType);
-                finishAffinity();
```
Comment (iaulakh) [inline AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/SetupActivity.java:244]: sent PR to main to remove this line

File: AzureCalling/app/src/main/AndroidManifest.xml [hunk 11]
```diff
@@ -60,15 +60,13 @@
             android:name="com.azure.samples.communication.calling.activities.SetupActivity"
             android:exported="true"
             android:label="@string/title_activity_setup"
-            android:taskAffinity="com.azure.samples.communication.calling.setupaffinity"
```
[the remark this answers is not in the document]
Comment (iaulakh) [reply AzureCalling/app/src/main/AndroidManifest.xml]: done

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java (hunk 1 above)
Comment (iaulakh) [reply AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/CallActivity.java]: done

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/IntroActivity.java [hunk 12]
```diff
@@ -18,8 +18,8 @@
 import com.azure.samples.communication.calling.AzureCalling;
 import com.azure.samples.communication.calling.R;
 import com.azure.samples.communication.calling.helpers.AppSettings;
-import com.azure.samples.communication.calling.helpers.Constants;
 import com.azure.samples.communication.calling.helpers.JoinCallType;
+import com.azure.samples.communication.calling.helpers.Constants;
```
Comment (waynemo) [inline AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/IntroActivity.java:22]: I think we can revert this.

File: AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/SetupActivity.java [hunk 13]
```diff
@@ -123,6 +123,12 @@ protected void onResume() {
         }
     }
 
+    @Override
```
Comment (waynemo) [inline AzureCalling/app/src/main/java/com/azure/samples/communication/calling/activities/SetupActivity.java:126]: We can revert this change.

File: AzureCalling/app/src/main/res/layout-land/activity_call.xml [hunk 14]
```diff
@@ -11,40 +10,47 @@
...
+        app:layout_constraintTop_toTopOf="parent" />
+
     <androidx.constraintlayout.widget.ConstraintLayout
         android:id="@+id/previewHolder"
         android:layout_width="0dp"
         android:layout_height="0dp"
         app:layout_constraintBottom_toBottomOf="parent"
-        app:layout_constraintRight_toLeftOf="@+id/call_buttons"
         app:layout_constraintLeft_toLeftOf="parent"
+        app:layout_constraintRight_toLeftOf="@+id/call_buttons"
```
Comment (waynemo) [inline AzureCalling/app/src/main/res/layout-land/activity_call.xml:30]: We can move this line back to make merging easier.

File: AzureCalling/app/src/main/res/layout-land/activity_call.xml [hunk 15]
```diff
@@ -2,7 +2,6 @@
 <androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:app="http://schemas.android.com/apk/res-auto"
     xmlns:tools="http://schemas.android.com/tools"
-
```
Comment (waynemo) [inline AzureCalling/app/src/main/res/layout-land/activity_call.xml:5]: We can undo this line change to make merging easier.

File: AzureCalling/app/src/main/res/layout-land/activity_call.xml [hunk 16]
```diff
@@ -57,8 +63,8 @@
         android:gravity="center"
         android:orientation="vertical"
         app:layout_constraintBottom_toBottomOf="parent"
-        app:layout_constraintRight_toRightOf="parent"
         app:layout_constraintLeft_toRightOf="@id/previewHolder"
+        app:layout_constraintRight_toRightOf="parent"
```
Comment (waynemo) [inline AzureCalling/app/src/main/res/layout-land/activity_call.xml:67]: We can move this line back to make merging easier.

File: AzureCalling/app/src/main/res/layout-land/activity_call.xml [hunk 17]
```diff
@@ -130,29 +177,30 @@
...
             android:layout_width="wrap_content"
-            android:minWidth="400dp"
-            android:maxWidth="400dp"
             android:layout_height="52dp"
             android:layout_marginLeft="16dp"
             android:layout_marginRight="16dp"
             android:background="@drawable/button_filled"
             android:backgroundTint="@color/colorAccent"
+            android:maxWidth="400dp"
+            android:minWidth="400dp"
```
Comment (waynemo) [inline AzureCalling/app/src/main/res/layout-land/activity_call.xml:186]: We can move these lines back to make merging easier.

File: AzureCalling/app/src/main/res/layout-land/activity_call.xml [hunk 18]
```diff
@@ -130,29 +177,30 @@
...
+            android:layout_marginTop="13dp"
             android:layout_marginRight="16dp"
             android:background="@drawable/button_outlined_grey"
+            android:maxWidth="400dp"
+            android:minWidth="400dp"
             android:text="@string/call_hangup_cancel"
             android:textColor="@android:color/white" />
 
     </LinearLayout>
+
```
Comment (waynemo) [inline AzureCalling/app/src/main/res/layout-land/activity_call.xml:205]: We can move these lines back to make merging easier.

File: AzureCalling/app/src/main/res/layout/activity_call.xml [hunk 19]
```diff
@@ -113,48 +118,90 @@
...
             android:layout_marginLeft="16dp"
             android:layout_marginRight="16dp"
             android:layout_marginBottom="13dp"
             android:background="@drawable/button_filled"
             android:backgroundTint="@color/colorAccent"
-            android:text="@string/call_hangup_confirm"
             android:gravity="center"
+            android:maxWidth="400dp"
+            android:minWidth="400dp"
+            android:text="@string/call_hangup_confirm"
```
Comment (waynemo) [inline AzureCalling/app/src/main/res/layout/activity_call.xml:188]: We can move these lines back to make merging easier.

File: AzureCalling/app/src/main/res/layout/activity_call.xml [hunk 20]
```diff
@@ -113,48 +118,90 @@
...
             android:layout_marginBottom="23dp"
             android:background="@drawable/button_outlined_grey"
-            android:text="@string/call_hangup_cancel"
             android:gravity="center"
+            android:maxWidth="400dp"
+            android:minWidth="400dp"
+            android:text="@string/call_hangup_cancel"
             android:textColor="@android:color/white" />
     </LinearLayout>
+
```
Comment (waynemo) [inline AzureCalling/app/src/main/res/layout/activity_call.xml:206]: We can move these lines back to make merging easier.

File: AzureCalling/app/src/main/res/layout/activity_intro.xml [hunk 21]
```diff
@@ -99,6 +99,7 @@
         android:text="@string/intro_sign_out"
         android:textColor="@color/colorAccent"
         android:visibility="gone"/>
+
```
Comment (waynemo) [inline AzureCalling/app/src/main/res/layout/activity_intro.xml:102]: We can undo this to make merging easier.

File: AzureCalling/app/src/main/res/layout/participant_view.xml [hunk 22]
```diff
@@ -4,6 +4,7 @@
     android:layout_width="match_parent"
     android:layout_height="match_parent"
     android:orientation="vertical">
+
```
Comment (waynemo) [inline AzureCalling/app/src/main/res/layout/participant_view.xml:7]: We can undo this to make merging easier.

**Review by waynemo — CHANGES_REQUESTED** [review]

**Review by waynemo — APPROVED** [review]

**Review by pavelprystinka — APPROVED** [review]
